executor/index_lookup_merge_join.go

```diff
@@ -387,13 +390,17 @@ func (omw *outerMergeWorker) increaseBatchSize() {
 func (imw *innerMergeWorker) run(ctx context.Context, wg *sync.WaitGroup, cancelFunc context.CancelFunc) {
```

Comment:
It's better to use `util.WithRecovery()` to start this goroutine, thus you don't need to copy the code to print the stack.

You can log a message in the `recoverFn` to indicate that this worker is an inner worker of index merge join.

Reply:
`util.WithRecovery()` execute in the function which starts the goroutine. But in this case, we need to `close(task.results)`, which is defined in inner goroutine function. `util.WithRecovery()` can't do this.